test/mocha/40-database.js

```diff
@@ -0,0 +1,255 @@
+      async () => {
+        const {executionStats} = await brEdvStorage.edvs.find({
+          controller: edvConfig.controller,
+          query: {'config.id': mockEdvId},
+          explain: true
+        });
+        executionStats.nReturned.should.equal(1);
+        executionStats.totalKeysExamined.should.equal(1);
+        executionStats.totalDocsExamined.should.equal(1);
+        executionStats.executionStages.inputStage.stage
+          .should.equal('IXSCAN');
+      });
+    it(`is properly indexed for 'config.id' and 'config.sequence' in ` +
+      'update()', async () => {
+      edvConfig.sequence += 1;
+      const {executionStats} = await brEdvStorage.edvs.update({
+        config: edvConfig,
+        explain: true
+      });
+      executionStats.nReturned.should.equal(1);
+      executionStats.totalKeysExamined.should.equal(1);
+      executionStats.totalDocsExamined.should.equal(1);
+      executionStats.executionStages.inputStage.inputStage.stage
+        .should.equal('IXSCAN');
+    });
+    it(`is properly indexed for 'config.id' in get()`, async () => {
+      const {executionStats} = await brEdvStorage.edvs.get({
+        id: edvConfig.id,
+        explain: true
+      });
+      executionStats.nReturned.should.equal(1);
+      executionStats.totalKeysExamined.should.equal(1);
+      executionStats.totalDocsExamined.should.equal(1);
+      executionStats.executionStages.inputStage.inputStage.inputStage.stage
+        .should.equal('IXSCAN');
+    });
+  });
+});
+
+describe('Chunks Database Tests', () => {
+  describe('Indexes', async () => {
+    let doc;
+    let chunk;
+    beforeEach(async () => {
+      await helpers.prepareDatabase();
+      const {methodFor} = await didKeyDriver.generate();
+      keyAgreementKey = methodFor({purpose: 'keyAgreement'});
+      kid = keyAgreementKey.id;
+
+      const {doc2} = mockData;
+      doc = {...doc2};
+      doc.jwe.recipients[0].header.kid = kid;
+      const docInsertResult = await brEdvStorage.docs.insert({
+        edvId: mockEdvId,
+        doc
+      });
+      const {doc: {jwe}} = docInsertResult;
+
+      const data = helpers.getRandomUint8();
+      const reader = await helpers.createEncryptStream(
+        {recipients: jwe.recipients, chunkSize, data});
+      const {value} = await reader.read();
+
+      // inserts chunk into database
+      chunk = {
+        sequence: doc.sequence,
+        ...value,
+      };
+      await brEdvStorage.chunks.update(
+        {edvId: mockEdvId, docId: doc.id, chunk});
+    });
+    it(`is properly indexed for 'localEdvId', 'docId' and 'chunk.index' in ` +
+      'update()', async () => {
+      const {executionStats} = await brEdvStorage.chunks.update({
+        edvId: mockEdvId,
+        docId: doc.id,
+        chunk,
+        explain: true
+      });
+      executionStats.nReturned.should.equal(1);
+      executionStats.totalKeysExamined.should.equal(1);
+      executionStats.totalDocsExamined.should.equal(1);
```

Comment:
I have a general concern around these assertions about total keys/docs examined throughout this test suite (and it may apply in other modules where this same pattern was used). In short, if we aren't making sure that more than one record is inserted prior to running these tests, then these assertions aren't really doing their job, i.e., if there is only ever one record in the database when this is run, then it total keys/docs examined can only ever be 1. These tests should make sure the number of records is more than 1.

Reply:
Well the main important check here is the `executionStats.executionStages.inputStage.stage.should.equal('IXSCAN');` since ultimately, that is what the tests are checking for. I can go ahead an update all these tests are all other tests in other libraries as well to insert multiple test records. These numbers would only not be 1 if there are some non unique indexes used, which there are in a couple cases such as the `config.controller` in the `edv.js` which is used in the `find()` query in this library.

Reply:
Yeah, I think it would be best if we made these assertions matter by adding mock records like you suggest.

Reply:
Ok will do.